internal/testutil/acc/config_formatter_test.go

```diff
@@ -104,3 +108,257 @@ func TestFormatToHCLLifecycleIgnore(t *testing.T) {
+      }
+    }
+  }
+
+  depends_on = [mongodbatlas_private_endpoint_regional_mode.atlasrm, mongodbatlas_privatelink_endpoint_service.atlasple]
+}
+`
+var twoReplicationSpecs = `
+resource "mongodbatlas_advanced_cluster" "cluster_info" {
+  backup_enabled = false
+  cluster_type   = "REPLICASET"
+  name           = "my-name"
+  project_id     = "project"
+
+  replication_specs {
+    num_shards = 1
+    zone_name  = "Zone 1"
+
+    region_configs {
+      priority      = 7
+      provider_name = "AWS"
+      region_name   = "US_WEST_1"
+      auto_scaling {
+        disk_gb_enabled = false
+      }
+      electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+      }
+    }
+  }
+  replication_specs {
+    num_shards = 1
+    zone_name  = "Zone 2"
+
+    region_configs {
+      priority      = 7
+      provider_name = "AWS"
+      region_name   = "EU_WEST_2"
+      auto_scaling {
+        disk_gb_enabled = false
+      }
+      electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+      }
+    }
+  }
+
+}
+`
+var twoRegionConfigs = `
+resource "mongodbatlas_advanced_cluster" "cluster_info" {
+  backup_enabled = false
+  cluster_type   = "REPLICASET"
+  name           = "my-name"
+  project_id     = "project"
+
+  replication_specs {
+    num_shards = 1
+    zone_name  = "Zone 1"
+
+    region_configs {
+      priority      = 7
+      provider_name = "AWS"
+      region_name   = "US_WEST_1"
+      auto_scaling {
+        disk_gb_enabled = false
+      }
+      electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+      }
+    }
+
+    region_configs {
+      priority      = 7
+      provider_name = "AWS"
+      region_name   = "EU_WEST_1"
+      auto_scaling {
+        disk_gb_enabled = false
+      }
+      electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+      }
+    }
+  }
+
+}
+`
+
+func Test_ClusterResourceHcl(t *testing.T) {
+	var (
+		clusterName                 = "my-name"
+		replicationSpecMultiRegions = admin.ReplicationSpec{
+			NumShards: conversion.IntPtr(1),
+			ZoneName:  conversion.StringPtr("Zone 1"),
+			RegionConfigs: &[]admin.CloudRegionConfig{
+				acc.CloudRegionConfig("US_WEST_1", constant.AWS, "M10", 3),
+				acc.CloudRegionConfig("EU_WEST_1", constant.AWS, "M10", 3),
+			},
+		}
+		testCases = map[string]struct {
+			expected string
+			req      acc.ClusterRequest
+		}{
+			"defaults": {
+				standardClusterResource,
+				acc.ClusterRequest{ClusterNameExplicit: clusterName},
+			},
+			"dependsOn": {
+				dependsOnClusterResource,
+				acc.ClusterRequest{ClusterNameExplicit: clusterName, ResourceDependencyName: "mongodbatlas_project.project_execution"},
+			},
+			"dependsOnMulti": {
+				dependsOnMultiResource,
+				acc.ClusterRequest{ClusterNameExplicit: clusterName, ResourceDependencyName: fmt.Sprintf("%s, %s", "mongodbatlas_private_endpoint_regional_mode.atlasrm", "mongodbatlas_privatelink_endpoint_service.atlasple")},
```

Comment:
as this is a test example, i would just say: 
```
ResourceDependencyName: "mongodbatlas_private_endpoint_regional_mode.atlasrm, mongodbatlas_privatelink_endpoint_service.atlasple"
```

Reply:
changed in bfb2a5e7939df716a73f6c3d4ee661214082a21f